libs/api_common/CMakeLists.txt

```diff
@@ -51,6 +52,7 @@ target_link_libraries("api_common" PRIVATE
 target_compile_definitions("api_common" PRIVATE
   _WINDOWS
   _USRDLL
+  USER_MODE
```

Comment:
I would hope that the USER_MODE define can be deleted in the future.  But since it was there before thus PR, it's fine in this PR

Reply:
I thought about this too... one way would be to reverse the logic, so that the default is UM, i.e. replace USER_MODE with a new KERNEL_MODE.

Reply:
I added a new `KERNEL_MODE` preprocessor variable in the required projects, and rebased the eBPF store logic on that. This way, the user can just consume the header/lib without any #define awareness.